### PR TITLE
Feature/36 GitHub actions

### LINKED ISSUE #36: Create GitHub Actions
**Summary.**
Create the GitHub actions file to run on every Pull Request and run the following:
- Flake8 
- Unit tests

### PR DESCRIPTION
**Pull request by DorianMoreno**
### Summary
Added GitHub actions to execute Linter and tests when a Pull Request is created.

### Test Plan
None yet

### Related Issues
close #36

### CODE REVIEW

**Review by camiloserr — APPROVED** [review]
re bien👍🏻

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
+name: Lint
+
+on:
+  push:
+    branches:
+      - master
+    paths:
+      - '*.py'
```
Comment (juanpa097) [inline .github/workflows/lint.yml]: Why do we add this lines? If I'm not wrong this triggers the action if we add a `.py` file in the root of the folder. I think that if we are gonna use this we should change it for `**.py` to trigger the build anytime you push a Python file, but I don't think this is necessary as we should run the action for every PR.

File: .github/workflows/lint.yml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+          python-version: 3.7.7
+          architecture: x64
+      - name: Checkout PyTorch
+        uses: actions/checkout@master
+      - name: Install flake8
+        run: pip install flake8
+      - name: Run flake8
+        uses: suo/flake8-github-action@releases/v1
+        with:
+          checkName: 'flake8_py3'   # NOTE: this needs to be the same as the job name
```
Comment (juanpa097) [inline .github/workflows/lint.yml]: I see here that we are relaying on a third party to run `flake8`. I don't think this is necessary as we chan make this change to run `flake8`:
```suggestion
      - name: Run flake8
      - run: flake8 .
```

I'm not sure if this works completely but is something similar in order to run the command and not relaying in third parties.

File: .github/workflows/lint.yml [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (juanpa097) [inline .github/workflows/lint.yml]: I don't think we need any token for this action. 
```suggestion
```

File: .github/workflows/tests.yml [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+        with:
+          python-version: 3.7.7
+          architecture: x64
+      - name: Install dependencies
+        run: |
+          python -m pip install -e .
+          python -m pip install -r ./requirements.txt
+      - name: Run tests
+        run: |
+          python ./tests.py
```
Comment (juanpa097) [inline .github/workflows/tests.yml]: I'm not sure, but I think the command for running the tests is:
```suggestion
        run:  python manage.py test
```
I don't know if `python ./tests.py` works too.
@johan-smc Can confirm this and provide us with the right command.

**Review by juanpa097 — DISMISSED** [review]
Amazing job and thanks for the initiative!

I left a few comments for us to discuss. I think that we should trigger both actions in the same circumstances so the `on:` section could be the same for both actions.
I saw that the actions were not running properly so we cold take a look at that. [See more info here](https://github.com/CapituloJaverianoACM/WebSiteAPI/runs/832291547?check_suite_focus=true)

File: .github/workflows/tests.yml (hunk 4 above)
Comment (DorianMoreno) [reply .github/workflows/tests.yml]: I saw projects running tests both ways, however I'm sure it depends on the tests. As there are no tests yet, I just took the liberty of choosing that option.

File: .github/workflows/lint.yml (hunk 1 above)
Comment (DorianMoreno) [reply .github/workflows/lint.yml]: Yes, you're right. I am going to experiment a little with these changes, so I will make a lot of commits until I get it right and the check goes through.

**Comment by DorianMoreno** [comment]
Ok, so I managed to fix `flake8` actions but checks are not passing because of some excepts. I created a new Issue (#45) to address that particular problem and focus this PR solely on getting actions working.

However, the tests action has presented other problems that I'm working on. Si this PR is still on hold.

**Comment by DorianMoreno** [comment]
Finally, after years of testing and trying things I found the error (wasn't checking out the branch and therefore couldn't access the files in the branch). Now both checks are running correctly, however the project is not passing them. The reasons for the tests not passing are already addressed in the issues #45 (What holds `flake8` check) and #48 (What holds `Test` check).

These issues were not fixed in this PR because they fall out of its scope and we should rejoice in the working checks.

**Review by BeltranDaniel — APPROVED** [review]
Good job

**Pull request merged**